R/rank_shift.R

```diff
@@ -76,32 +76,38 @@ mean_rank_shift <- function(df, time.var = "year", species.var = "species",
 #' @param abundance.var The abundance variable
 #' @return a dataframe, showing years compared
 meanrank <- function(comm_data, time.var = "year",
-                     species.var = "species", abundance.var = "abundance") {
+                     species.var = "species",
+                     abundance.var = "abundance") {
     ## split data by year
-    yearlist <- split(comm_data, comm_data[[time.var]])
-    ## Compare consecutive pairs of years
-    y1 <- yearlist[-length(yearlist)]
-    y2 <- yearlist[-1]
 
-    commonspp <- Map(df_intersect, y1, y2, dataname = species.var)
+  for (v in c(time.var, species.var, abundance.var)){
+    assertthat::assert_that(assertthat::has_name(comm_data, v))
```

Comment:
assert that those names are actually present in the data (very weird errors if they are not!)